### PR TITLE
[Bug] Fix IVF merge sort when refine factor is presented.

### PR DESCRIPTION
**Pull request by eddyxu**
Previously, at the merge sort stage, IVF does not returns k * refine_factor vectors for the Take node to load the original data. This fix makes IVF returns `k * refine_factor` vectors from `IVF::search`

### CODE REVIEW

File: .github/workflows/python.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ on:
       - .github/workflows/run_tests/**
 jobs:
   linux:
-    timeout-minutes: 30
+    timeout-minutes: 45
```
Comment (changhiskhan) [inline .github/workflows/python.yml:18]: i thought the long poll is duckdb and not the python gha?

File: rust/src/index/vector/opq.rs [hunk 2]
```diff
@@ -108,7 +110,7 @@ impl OptimizedProductQuantizer {
         let dim = train.num_columns();
 
         // Train few times to get a better rotation matrix. See Faiss.
-        pq.train(&train, metric_type, 4).await?;
+        pq.train(&train, metric_type, 1).await?;
```
Comment (changhiskhan) [inline rust/src/index/vector/opq.rs:113]: why the change to only once?

File: rust/src/index/vector/opq.rs [hunk 3]
```diff
@@ -135,6 +137,14 @@ impl OptimizedProductQuantizer {
     }
 }
 
+/// Initialize rotation matrix.
+fn init_rotation(dimension: usize) -> Result<MatrixView> {
+    let mat = MatrixView::random(dimension, dimension);
+    let (u, _, vt) = mat.svd()?;
```
Comment (changhiskhan) [inline rust/src/index/vector/opq.rs:143]: here we're using SVD in-place of FAISS's QR thing right?

File: rust/src/index/vector/ivf.rs [hunk 4]
```diff
@@ -133,10 +133,12 @@ impl VectorIndex for IVFIndex {
                 batch.schema()
             ))
         })?;
-        let refined_index = sort_to_indices(score_col, None, Some(query.k))?;
 
+        // TODO: Use a heap sort to get the top-k.
+        let limit = query.k * query.refine_factor.unwrap_or(1) as usize;
```
Comment (changhiskhan) [inline rust/src/index/vector/ivf.rs:138]: this is the fix right?

File: rust/src/index/vector/opq.rs (hunk 3 above)
Comment (eddyxu) [reply rust/src/index/vector/opq.rs:143]: yes

File: rust/src/index/vector/ivf.rs (hunk 4 above)
Comment (eddyxu) [reply rust/src/index/vector/ivf.rs:138]: Yes.

File: rust/src/index/vector/opq.rs (hunk 2 above)
Comment (eddyxu) [reply rust/src/index/vector/opq.rs:113]: Tested a few times , dont see major perf difference at the moment, but speed up OPQ training a bit.

File: .github/workflows/python.yml (hunk 1 above)
Comment (eddyxu) [reply .github/workflows/python.yml:18]: There are a few create index tests seem slow on GHA.
https://github.com/eto-ai/lance/actions/runs/4493433922

**Review by changhiskhan — APPROVED** [review]
